### ISSUE TITLE
Operator support for HA deployments

### ISSUE THREAD

**Issue by pmorie** [knative/serving-operator#279]
**Problem**
We are currently working on making knative controllers able to do [leader-elected HA](knative/pkg#1007)

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
People using the operator to deploy serving (and possibly other parts of the project if we unify the operators), who are making decisions about how they want their knative deployment configured.

**Exit Criteria**
I should be able to indicate via the operator's CR that I want:

- Leader election disabled / enabled
- Number of replicas for controllers that scale horizontally
- Possibly anti-affinity
- Possibly disruption budgets

**Time Estimate (optional):**
Probably more dependent on design consensus than implementation time.

**Comment by jcrossley3** [comment]
It's easy enough to conceive of a new field in the CR...
```yaml
spec:
  leader-election:
    - enabled: true 
      replicas: 4
```
...but if all the operator does in response is set ConfigMap entries as described in the design doc, then we may as well use the existing *Spec.Config* field instead.

**Comment by jcrossley3** [comment]
Of course, if we make the controllers consume the `KnativeServing` CR for their leader-election configuration instead of a `ConfigMap`, that may be a good first step toward migrating other controller configuration from their `ConfigMaps` to the `KnativeServing` CR.

Maybe that's your master plan. :stuck_out_tongue_winking_eye:

**Comment by pmorie** [comment]
I'd want to avoid serving depending on the operator's CR - it's the operator's job to make sure serving does what's in the CR.

I think the operator will wind up needing to do more than just manipulate configmaps, because there will likely be knobs that affect the resources deployed (replicas, anti-affinity etc) too.

**Comment by pmorie** [comment]
As I write this I'm thinking maybe the operator API needs to be in the design doc for this feature track

**Comment by aliok** [comment]
@pmorie 
Can we resolve this ticket as following PRs are all merged, or is there anything missing?

https://github.com/knative/serving-operator/pull/321
https://github.com/knative/serving-operator/pull/326
https://github.com/knative/serving-operator/pull/337
